projects/step-frontend/src/lib/modules/timeseries/execution-page/execution-page.component.ts

```diff
@@ -1,24 +1,42 @@
 import { Component, Input, OnDestroy, OnInit, ViewChild } from '@angular/core';
-import { AJS_MODULE, AsyncTasksService, DashboardService, ExecutionsService, pollAsyncTask } from '@exense/step-core';
+import {
+  AJS_MODULE,
+  AsyncTasksService,
+  DashboardService,
+  Execution,
+  ExecutionsService,
+  pollAsyncTask,
+} from '@exense/step-core';
 import { downgradeComponent, getAngularJSGlobal } from '@angular/upgrade/static';
 import { PerformanceViewSettings } from '../performance-view/model/performance-view-settings';
 import { TimeSeriesService } from '../time-series.service';
 import { TimeSeriesContextsFactory } from '../time-series-contexts-factory.service';
 import { RangeSelectionType } from '../time-selection/model/range-selection-type';
 import { PerformanceViewComponent } from '../performance-view/performance-view.component';
-import { forkJoin, Observable, of, Subject, Subscription, switchMap, tap, timer } from 'rxjs';
+import { forkJoin, Observable, of, Subject, Subscription, switchMap, takeUntil, tap, timer } from 'rxjs';
+import { RelativeTimeSelection } from '../time-selection/model/relative-time-selection';
+import { TimeRangePickerSelection } from '../time-selection/time-range-picker-selection';
+import { TSTimeRange } from '../chart/model/ts-time-range';
+import { TimeSeriesContext } from '../time-series-context';
+import { TimeSeriesUtils } from '../time-series-utils';
 
 @Component({
   selector: 'step-execution-performance',
   templateUrl: './execution-page.component.html',
   styleUrls: ['./execution-page.component.scss'],
 })
 export class ExecutionPageComponent implements OnInit, OnDestroy {
-  private readonly RUNNING_EXECUTION_END_TIME_BUFFER = 5000; // if the exec is running, we don't grab the last 5 seconds
+  readonly ONE_HOUR_MS = 3600 * 1000;
+
+  terminator$ = new Subject<void>();
 
   @ViewChild(PerformanceViewComponent) performanceView!: PerformanceViewComponent;
 
   @Input() executionId!: string;
+  private execution: Execution | undefined;
```

Comment:
```suggestion
  private execution?: Execution;
```
Should be the same in this case